src/zjs_spi.c

```diff
@@ -93,6 +93,13 @@ static ZJS_DECL_FUNC(zjs_spi_transceive)
         }
     }
 
+    // If a buffer is provided, check that it is a valid type
+    if (argc > 1 && (!jerry_value_is_array(argv[1]) &&
+                      !jerry_value_is_string(argv[1]) &&
+                      !zjs_value_is_buffer(argv[1]))) {
+        return SYNTAX_ERROR("SPI transceive buffer is an invalid type");
+    }
```

Comment:
This should not be needed - because of the ZJS_VALIDATE_ARGS call above, if you have an argv[1] it will have to be one of these three types specified there.